### PR TITLE
Initialize project with components

### PR DESCRIPTION
**Pull request by MostafaShahat2021**
## In this PR:
- Make sure that there are [no linter errors](https://github.com/microverseinc/linters-config).
- Make sure that I used the correct [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
 - Make sure that I documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
 - Follow Microverse's list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).
- ⭐️Setup a new React application using ```create-react-app```.
- ⭐️Create a directory for reusable components: ```/components```.
- ⭐️Install [React Router V6](https://reactrouter.com/en/main/start/tutorial#setup)
- ⭐️ Setup router inside of ```<App>```
- ⭐️Add 2 routes in the application:
   - ```/```
   - ```/categories```
 - ⭐️Create components inside of the ```/components``` directory:
   - Contains individual book state, as well as a button to delete the book 
   - Contains a list to render individual books
   - Contains a form to create new books
   - Contains a navigation to link to the 2 routes you created

### CODE REVIEW

File: src/components/BookList.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+//     <div>
+//       <h2>Book List</h2>
+//       <div>
+//         <Book title={title} author={author} onClick={handelDelete} />
+//       </div>
+//     </div>
+//   );
+// }
+
+// export default BookList;
```
Comment (Efiamotu-1) [inline src/components/BookList.js]: - [x] It is unprofessional to leave commented code in your code base so I kindly suggest you remove them 👍

File: src/components/BookList.js
Comment (Efiamotu-1) [inline src/components/BookList.js:1]: - [x] This Booklist component is to render a list of books using the `<Book>` component
- [x] You are to also have a `<Books>` component which holds the `<Booklist> & <Book>` component

**Review by Efiamotu-1 — CHANGES_REQUESTED** [review]
Hi @MostafaShahat2021 , 

 

Good job so far! 👏. Your Code is Awesome! 💪

There are some **few** issues 🐛 that you still need to work 🔧 on though to go to the next project ⏭️ but you are almost there! 🏃
## Don't Worry 🙏

It's ok to have bugs in your code 🐛. Please, see this code review as a **chance** for you to **learn** practical coding skills and learn **new tips** that can improve your code quality 💯 and help you **get a better ** 💼.

_I am going to guide you on how to fix them down below. 🤓_

## Highlights :dart: 
- ✔️ No linter Errors
- ✔️ Descriptive README
- ✔️ Descriptive PR title and description
- The code is top-notch! 💯 and has a huge potential.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, don't Merge This Pull Request yet. Only Merge when it is Approved ✅.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/BookList.js (hunk 1 above)
Comment (MostafaShahat2021) [reply src/components/BookList.js]: Hi @Efiamotu-1, Sorry I'll remove the unused code

File: src/components/BookList.js
Comment (MostafaShahat2021) [reply src/components/BookList.js:1]: Thanks a lot for your suggestion, I'll fix that 😄 .

**Review by Felix45 — APPROVED** [review]
# Approved :green_circle: 
Hi @MostafaShahat2021, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights :heavy_check_mark: 
- [x] BookList component created correctly :white_check_mark: 
- [x] Git flow is used correctly :white_check_mark: 
- [x] Pull request has a descriptive summary and title :white_check_mark:   
- [x] Implemented all the changes from the previous reviewer :white_check_mark:  

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by MostafaShahat2021** [comment]
Hi @Felix45  👋 , Thanks a lot :100:  for your efforts in reviewing my code, I really appreciate it 😀 .

**Pull request merged**